subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -605,11 +626,132 @@ void end_report(void)
 K_APPMEM_PARTITION_DEFINE(ztest_mem_partition);
 #endif
 
+static void __ztest_init_unit_test_result_for_suite(struct ztest_suite_node *suite)
+{
+	struct ztest_unit_test *test = NULL;
+
+	while (((test = z_ztest_get_next_test(suite->name, test)) != NULL)) {
+		test->stats->run_count = 0;
+		test->stats->skip_count = 0;
+		test->stats->fail_count = 0;
+		test->stats->pass_count = 0;
+	}
+}
+
+static void flush_log(void)
+{
+	if (IS_ENABLED(CONFIG_LOG_PROCESS_THREAD)) {
+		while (log_data_pending()) {
+			k_sleep(K_MSEC(10));
+		}
+		k_sleep(K_MSEC(10));
+	} else {
+		while (LOG_PROCESS()) {
+		}
+	}
+}
+
+/* Show one line summary for a test suite.
+ */
+static void __ztest_show_suite_summary_oneline(struct ztest_suite_node *suite)
+{
+	int distinct_pass = 0, distinct_fail = 0, distinct_skip = 0, distinct_total = 0;
+	int passrate = 0, passrate1 = 0, passrate2 = 0;
+	int suite_result = TC_PASS;
+
+	struct ztest_unit_test *test = NULL;
+
+	/** summary of disctinct run  */
+	while (((test = z_ztest_get_next_test(suite->name, test)) != NULL)) {
+		distinct_total++;
+		if (test->stats->skip_count == test->stats->run_count) {
+			distinct_skip++;
+		} else if (test->stats->pass_count == test->stats->run_count) {
+			distinct_pass++;
+		} else {
+			distinct_fail++;
+		}
+	}
+
+	if (distinct_skip == distinct_total) {
+		suite_result = TC_SKIP;
+		passrate1 = passrate2 = 0;
+	} else {
+		suite_result = (distinct_fail > 0) ? TC_FAIL : TC_PASS;
+		passrate = distinct_pass * 100;
+		passrate1 = passrate / (distinct_total - distinct_skip);
+		passrate2 = (passrate - passrate1 * (distinct_total - distinct_skip)) / 100;
```

Comment:
Please double check my algebra on this: But I think passrate2 is always equal to zero here.

`passrate` == `passrate1 * (distinct_total - distinct_skip)`

Reply:
All `passrate, passrate1, passrate2` are of int type. The line 693 is integer division.
For example: suppose `distinct_pass=1234`, `distinct_total - distinct_skip = 10000`, then `passrate=1234*100=123400`, `passrate1= 123400/10000 = 12`, `passrate2 = (123400 - 12*10000)/100 = 34`.
The final output on line 697 will be ` 12.34%`.

Reply:
At first I thought this was just integer division and tried with below:
Suppose `distinct_pass=1234`, `distinct_total - distinct_skip = 10000`,
then `passrate=1234*100=123400`, 
`passrate1= 123400/10000 = 12`,
`passrate2 = (123400 - 12*10000)/100 = 34`.
The final output on line 697 will be `  12.34%`.

But it turns out not fully correct for other scenarios. Such as `2/3=66.67%`.
And I didn't consider the rounding in the last implementation.

Thanks for pointing this out! I have refined it. Could you have another review.
The new calculation is like this:
Say:
```
distinct_total=3;
distinct_pass=2;
distinct_skip=distinct_fail=0;
```
Then:
```
essential_total=3-0=3;
passrate = 2*100000 = 200000
passrate0 = 200000/3 = 66666
passrate1 = 66666 / 1000 = 66
passrate2 = (66666 -  66*1000)/10=66
passrate3 = 66666 - 66*1000 - 66*10 = 6
passrate3 >=5, so passrate2++
```
At last we got: `passrate1.passrate2% =  66.67%`

Reply:
Ahh I understand more now. It'd help if we renamed the passrate variables with more descriptive terms representing their intermediary values though. It'd make it easier for the next person wanting to look at or change this code for whatever reason.

Reply:
Indeed, I don't like those variable names either. English is not my native language but I will try. It's like this now:

https://github.com/smrtos/zephyr/blob/becee84476b3606e7912178c0fea459c9ba77330/subsys/testsuite/ztest/src/ztest_new.c#L694-L702